compose/ogmios/Dockerfile

```diff
@@ -0,0 +1,77 @@
+  --extra-conf "sandbox = false" \
+  --extra-conf "substituters = https://cache.nixos.org https://cache.iog.io" \
+  --extra-conf "trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=" \
+  --init none --no-confirm
+ENV PATH="${PATH}:/nix/var/nix/profiles/default/bin"
+RUN mkdir -p /app
+WORKDIR /app
+ARG OGMIOS_VERSION
+ARG CARDANO_NODE_VERSION
+RUN cd /app &&\
+  git clone --recursive https://github.com/CardanoSolutions/ogmios.git ogmios-src &&\
+  cd ogmios-src &&\
+  git fetch --all --tags &&\
+  git checkout ${OGMIOS_VERSION}
+RUN echo >/app/default.nix $'\n\
+  let\n\
+    nodeFlake = builtins.getFlake "github:input-output-hk/cardano-node/'"${CARDANO_NODE_VERSION}"$'";\n\
+    system = "'"$([ "$TARGETPLATFORM" = "linux/arm64" ] && echo "aarch64-linux" || echo "x86_64-linux")"$'";\n\
+    inherit (nodeFlake.legacyPackages.${system}) haskell-nix;\n\
+    project = haskell-nix.project {\n\
+      compiler-nix-name = "ghc8107";\n\
+      projectFileName = "cabal.project";\n\
+      inputMap = { "https://input-output-hk.github.io/cardano-haskell-packages" = nodeFlake.inputs.CHaP; };\n\
+      src = haskell-nix.haskellLib.cleanSourceWith {\n\
+        name = "ogmios-src";\n\
+        src = ./ogmios-src;\n\
+        subDir = "server";\n\
+        filter = path: type: baseNameOf path != "package.yaml";\n\
+      };\n\
+      modules = [ ({ lib, pkgs, ... }: {\n\
+        packages.cardano-crypto-praos.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf ] ];\n\
+        packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 pkgs.libblst ] ];\n\
+      }) ];\n\
+    };\n\
+  in {\n\
+    ogmios = project.projectCross.musl64.hsPkgs.ogmios.components.exes.ogmios;\n\
+    cardano-node = nodeFlake.legacyPackages.${system}.hydraJobs.musl.cardano-node;\n\
+  }\n\
+  ' && cat /app/default.nix
+RUN nix-build /app/default.nix -A ogmios -o /app/ogmios
+# Note: ‘inputoutput/cardano-node’ doesn’t have all commits published, but we want to allow specifying
+#       any commit, so we have to build from source.
+RUN nix-build /app/default.nix -A cardano-node -o /app/cardano-node
+RUN ls -alh /app/ogmios/bin
+RUN ls -alh /app/cardano-node/bin
+RUN /app/ogmios/bin/ogmios --help
+RUN /app/cardano-node/bin/cardano-node --version
+
+FROM --platform=${TARGETPLATFORM} alpine:3.18 as cardano-node-ogmios
+RUN apk update && apk add bash tini && rm -rf /var/cache/apk/*
+COPY --from=haskell-builder /app/ogmios/bin/ogmios /bin/ogmios
+COPY --from=haskell-builder /app/cardano-node/bin/cardano-node /bin/cardano-node
+RUN cardano-node --version && ogmios --version  # check that they are indeed statically linked
+ARG NETWORK
+LABEL name=cardano-node-ogmios
+LABEL description="A Cardano node, side-by-side with its JSON WebSocket bridge."
+COPY --from=haskell-builder /app/ogmios-src/server/config/network/${NETWORK} /config
```

Comment:
Let's clone the source directly to have full control, from https://github.com/input-output-hk/cardano-configurations and add a build arg to allow a revision override. Can use `65ef979cf69f420efca0a7aaf0412a610bc48097` as a default

Reply:
Okay! :memo:

Reply:
Fixed in 0df1631dfa3a679792ed90471bc753d86394fcbc

Reply:
Approved. Please squash those two commits, then I'll test the build

Reply:
@rhyslbw done :+1: